### ISSUE TITLE
Master is failing to create the feedstocks

### ISSUE THREAD

**Issue by ocefpaf** [conda-forge/staged-recipes#221]
```
CI support files regenerated. These need to be pushed to github!
[master 6f53a0b] Re-render the feedstock after CI registration.
 2 files changed, 7 insertions(+), 7 deletions(-)
 rewrite conda-forge.yml (85%)
rm 'recipes/check/build.sh'
rm 'recipes/check/meta.yaml'
CI Summary for conda-forge/libmo_unpack-feedstock (can take ~30s):
 * conda-forge/libmo_unpack-feedstock already enabled on travis-ci
 * conda-forge/libmo_unpack-feedstock already enabled on CircleCI
 * conda-forge/libmo_unpack-feedstock already enabled on appveyor
Traceback (most recent call last):
  File "/Users/travis/miniconda/bin/conda-smithy", line 9, in <module>
    load_entry_point('conda-smithy==0.7.1', 'console_scripts', 'conda-smithy')()
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 227, in main
    args.subcommand_func(args)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_smithy/cli.py", line 162, in __call__
    ci_register.appveyor_configure(owner, repo)
  File "/Users/travis/miniconda/lib/python2.7/site-packages/conda_smithy/ci_register.py", line 114, in appveyor_configure
    raise ValueError(response)
ValueError: <Response [404]>
Traceback (most recent call last):
  File ".CI/create_feedstocks.py", line 123, in <module>
    subprocess.check_call(['conda', 'smithy', 'register-ci', '--feedstock_directory', feedstock_dir] + owner_info)
  File "/Users/travis/miniconda/lib/python2.7/subprocess.py", line 540, in check_call
    raise CalledProcessError(retcode, cmd)
subprocess.CalledProcessError: Command '['conda', 'smithy', 'register-ci', '--feedstock_directory', '/var/folders/gw/_2jq29095y7b__wtby9dg_5h0000gn/T/tmpmtorVK__feedstocks/libmo_unpack-feedstock', '--organization', 'conda-forge']' returned non-zero exit status 1
```

**Comment by jakirkham** [comment]
Pretty weird that it is always getting a 404 error.

**Comment by ocefpaf** [comment]
Let's hold off merging stuff until we figure this out.

**Comment by jakirkham** [comment]
Sorry shouldn't have closed.

**Comment by jakirkham** [comment]
Ugh, sorry, should still be open probably.

**Comment by ocefpaf** [comment]
@jakirkham I think I know what is going on and I am working on it. Please hold the merges a little bit.

**Comment by jakirkham** [comment]
Sure. That was the last one. I figured we should narrow down at least if that package was the problem. In the worst case, we can delete all its CIs and then the repo and try again.

**Comment by jakirkham** [comment]
Also, don't be surprised, but the feedstock update script is failing because the libmo_unpack repo is empty. Once it is fixed that problem will also be fixed.

**Comment by ocefpaf** [comment]
@jakirkham I need to get `libmo_unpack` dependencies to build first as its dependencies are fully formed feedstocks now. (`texinfo` in the first merge and `check` due to your delete/re-add of `libmo_unpack`.)

Once that is done I will delete a `libmo_unpack` project that is registered on AppVeyor and re-start the staged-recipes master build.  `libmo_unpack` skips AppVeyor, but I guess we register every project anyway.

**Comment by jakirkham** [comment]
> I need to get `libmo_unpack` dependencies to build first as its dependencies are fully formed feedstocks now. (`texinfo` in the first merge and `check` due to your delete/re-add of `libmo_unpack`.)

Sure. Should already be happening, no?

> Once that is done I will delete a `libmo_unpack` project that is registered on AppVeyor and re-start the staged-recipes master build.  `libmo_unpack` skips AppVeyor, but I guess we register every project anyway.

Ok. Sounds good.

**Comment by jakirkham** [comment]
Looks like `texinfo` and `check` are built now.

**Comment by ocefpaf** [comment]
![27569662d1455596983-x-transbots-mm-viii-arkose-beachcomber-image](https://cloud.githubusercontent.com/assets/950575/14110338/5839fffe-f59c-11e5-8f67-e70669c46e11.png)

**Comment by jakirkham** [comment]
Only letting you know. :wink:

**Comment by ocefpaf** [comment]
The dependencies were upload and the `libmo-unpack` build is working. But AppVeyor's registration is not.

The URL that is 404ing is `https://ci.appveyor.com/api/projects/{}/{}/settings` and that should resolve to https://ci.appveyor.com/api/projects/conda-forge/libmo-unpack-feedstock/settings, which is fine BTW.

However, something is amiss. And the problem is....

`libmo_unpack` vs `libmo-unpack`! I hate underscores and dash issues :unamused:

**Comment by jakirkham** [comment]
Ugh, I expected this to happen when I added something with an underscore before.

**Comment by jakirkham** [comment]
cc @pelson

**Comment by ocefpaf** [comment]
Not sure were to solve this though. Rename the package (hope not)? Make conda-smithy more robust to deal with this (probably yes)?

For now let's get this recipe out of the way and I will re-add it later.

**Comment by jakirkham** [comment]
Yeah, I think you're right. Sorry, we can't get it in yet. Maybe you can save a PR to re-add once we get this sorted.

**Comment by jakirkham** [comment]
Though I don't know about the underscore part because [this]( https://github.com/conda-forge/rank_filter-feedstock ) works. Has something changed since then?

**Comment by ocefpaf** [comment]
> Though I don't know about the underscore part because this works. Has something changed since then?

Yes. We are now editing AppVeyor's config using the API to skip builds when there is no `appveyor.yml`.

You'll notice that the API URL for `rank_filter` is: https://ci.appveyor.com/api/projects/conda-forge/rank-filter-feedstock/settings (note the `-` instead of `_`).

The feedstocks created before that change are fine because I edit the config manually.

**Comment by jakirkham** [comment]
Ah, ok. Thanks for clarifying.

**Comment by jakirkham** [comment]
Added this PR ( https://github.com/conda-forge/conda-forge.github.io/pull/66 ) so that we can update the feedstocks even though the [libmo_unpack feedstock]( https://github.com/conda-forge/libmo_unpack-feedstock ) is incomplete (i.e. has no commits).

**Comment by jakirkham** [comment]
Solved by this PR ( https://github.com/conda-forge/conda-smithy/pull/106 ), which was included in the `v0.7.2` release of conda-smithy.
